File: oldScratch/cpp/TileColor.cpp
#include "TileColor.h"
std::string TileColorEnum::consoleCode(TileColor color){
    if(color==TileColor::RED) return "\u001b[0;31;47m";
    if(color==TileColor::BLACK) return "\u001b[0;30;47m";
    if(color==TileColor::TEAL) return "\u001b[1;34;47m";
    if(color==TileColor::YELLOW) return "\u001b[1;33;47m";
    std::string err="unhandled color '";
    err+=char(color);
    err+="'";
    throw err;
}
const std::string TileColorEnum::ResetCode=  "\u001b[0m";
const std::string TileColorEnum::TileBG="\u001b[1;30;47m";
const std::array<TileColor,4> TileColorEnum::Values={{BLACK,RED,TEAL,YELLOW}};

File: RummikubSolver/SolverLogic/Models/TileColor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SolverLogic.Models
{
    public static class ConsoleColor
    {
        public static string ColorCode(this TileColor tileColor) => tileColor switch
        {
            TileColor.BLACK => "\u001b[0;30;100m",
            TileColor.RED => "\u001b[0;31m",
            TileColor.TEAL => "\u001b[1;34m",
            TileColor.YELLOW => "\u001b[1;33m",
            _ => throw new NotImplementedException($"no console code for {nameof(TileColor)} '{tileColor}'")
        };
        public static char Char(this TileColor tileColor) => tileColor.ToString()[0];

        public const string TileBackground = "\u001b[1;30;100m";
        public const string ResetCode = "\u001b[0m";
    }
    public enum TileColor
    {
        BLACK = 0,
        RED = 1,
        TEAL = 2,
        YELLOW = 3
    }
}
